### ISSUE TITLE
Custom LibmeshPartitioner doesn't work with displacements

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#7184]
### Description of the enhancement or error report

If you use the partitioner subblock in `[Mesh]` it fails to work if you use displacements.

### Rationale for the enhancement or information for reproducing the error

Add displacements to the `custom_linear_partitioner_tests` in moose/test

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

People using custom partitioners

**Comment by permcody** [comment]
Tagging @snschune

**Comment by snschune** [comment]
Is there a test that illustrates the defect already that I can simply run?

**Comment by permcody** [comment]
# Adaptivity on displaced problem
# - testing initial_refinement and adaptivity as well
#
# variables:
# - u and v_aux are used for displacing the problem
# - v is used to get some refinements
#

[Mesh]
  dim = 2
  type = GeneratedMesh
  nx = 3
  ny = 3

  xmin = 0.0
  xmax = 1.0

  ymin = 0.0
  ymax = 10.0


  uniform_refine = 2
  displacements = 'u aux_v'
  [./Partitioner]
   type = LibmeshPartitioner
   partitioner = linear
  [../]
 parallel_type = replicated
[]

[Functions]
  [./aux_v_fn]
    type = ParsedFunction
    value = x*(y-0.5)/5
  [../]
[]

[Variables]
  [./u]
    order = FIRST
    family = LAGRANGE
  [../]

  [./v]
    order = FIRST
    family = LAGRANGE
  [../]
[]

[Kernels]
  active = 'udiff uie vdiff  vie'

  [./udiff]
    type = Diffusion
    variable = u
  [../]

  [./uie]
    type = TimeDerivative
    variable = u
  [../]

  [./vdiff]
    type = Diffusion
    variable = v
  [../]

#   [./vconv]
#     type = Convection
#     variable = v
#     velocity = '-10 1 0'
#   [../]

  [./vie]
    type = TimeDerivative
    variable = v
  [../]
[]

[BCs]
  active = 'uleft uright vleft vright'

  [./uleft]
    type = DirichletBC
    variable = u
    boundary = 1
    value = 0
  [../]

  [./uright]
    type = DirichletBC
    variable = u
    boundary = 2
    value = 0.1
  [../]

  [./vleft]
    type = DirichletBC
    variable = v
    boundary = 1
    value = 1
  [../]

  [./vright]
    type = DirichletBC
    variable = v
    boundary = 2
    value = 0
  [../]
[]

[AuxVariables]
  [./aux_v]
    order = FIRST
    family = LAGRANGE
  [../]
[]

[AuxKernels]
  [./aux_k_1]
    type = FunctionAux
    variable = aux_v
    function = aux_v_fn
  [../]
[]

[Executioner]
  type = Transient
  solve_type = 'PJFNK'
  start_time = 0.0
  num_steps = 2
  dt = .1

  [./Adaptivity]
    refine_fraction = 0.2
    coarsen_fraction = 0.3
    max_h_level = 4
  [../]
[]

[Outputs]
  [./out]
    type = Exodus
    use_displaced = true
  [../]
[]

**Comment by snschune** [comment]
@permcody I cannot reproduce the error. How did you reproduce Maria's error?

**Comment by permcody** [comment]
Did you run it in parallel?

**Comment by snschune** [comment]
That did the trick. I'll look at it once I get time.

**Comment by snschune** [comment]
So here is what I've found quickly looking at this problem. I'll dig deeper but maybe @permcody or @jwpeterson know the answer right away. It appears that `MooseMesh::init` is called twice and the 
bool `_custom_partitioner_requested` is set differently. Is the second call for the displaced mesh?

I have changed the code in `MooseMesh::init` to print some info

```
void
MooseMesh::init()
{
  if (processor_id() == 0)
  {
    std::cout << "Entering Mesh::init on processor id " << processor_id() << " and custom partitioner is set to " << _custom_partitioner_requested << std::endl;
    libMesh::print_trace();
  }
  if (_custom_partitioner_requested)
  {
    mooseWarning("Processor id " << processor_id() << " has entered the custom partitioner branch. ");
    // Check of partitioner is supplied (not allowed if custom partitioner is used)
    if (!parameters().isParamSetByAddParam("partitioner"))
      mooseError("If partitioner block is provided, partitioner keyword cannot be used!");
    // Set custom partitioner
    if (!_custom_partitioner.get())
      mooseError("Custom partitioner requested but not set!");
    getMesh().partitioner().reset(_custom_partitioner.release());

  }
  else
  {
    mooseWarning("Processor id " << processor_id() << " has entered the partitioner keyword branch.");
    // Set standard partitioner
    // Set the partitioner based on partitioner name
```

That is the screen printout:

```
Entering the partitioner action
Leaving the partitioner action
Entering Mesh::init on processor id 0 and custom partitioner is set to 1


*** Warning ***
Processor id 1 has entered the custom partitioner branch. 
at /Users/schus/projects/moose/framework/src/mesh/MooseMesh.C, line 1789

Stack frames: 9
0: 0   libmesh_opt.0.dylib                 0x000000010719ae8d libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1533
1: 1   libmoose-opt.0.dylib                0x0000000106a395ac MooseMesh::init() + 236
2: 2   libmoose-opt.0.dylib                0x000000010674994c SetupMeshAction::act() + 92
3: 3   libmoose-opt.0.dylib                0x000000010670da5a ActionWarehouse::executeActionsWithAction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 714
4: 4   libmoose-opt.0.dylib                0x000000010670d77b ActionWarehouse::executeAllActions() + 123
5: 5   libmoose-opt.0.dylib                0x00000001068e34aa MooseApp::runInputFile() + 266
6: 6   libmoose-opt.0.dylib                0x00000001068e48af MooseApp::run() + 95
7: 7   moose_test-opt                      0x000000010639c512 main + 114
8: 8   libdyld.dylib                       0x00007fff916a25ad start + 1


*** Warning ***
Processor id 0 has entered the custom partitioner branch. 
at /Users/schus/projects/moose/framework/src/mesh/MooseMesh.C, line 1789

Entering Mesh::init on processor id 0 and custom partitioner is set to 0


*** Warning ***
Processor id 1 has entered the partitioner keyword branch.
at /Users/schus/projects/moose/framework/src/mesh/MooseMesh.C, line 1801

Stack frames: 9
0: 0   libmesh_opt.0.dylib                 0x000000010719ae8d libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1533
1: 1   libmoose-opt.0.dylib                0x0000000106a395ac MooseMesh::init() + 236
2: 2   libmoose-opt.0.dylib                0x0000000106749a64 SetupMeshAction::act() + 372
3: 3   libmoose-opt.0.dylib                0x000000010670da5a ActionWarehouse::executeActionsWithAction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 714
4: 4   libmoose-opt.0.dylib                0x000000010670d77b ActionWarehouse::executeAllActions() + 123
5: 5   libmoose-opt.0.dylib                0x00000001068e34aa MooseApp::runInputFile() + 266
6: 6   libmoose-opt.0.dylib                0x00000001068e48af MooseApp::run() + 95
7: 7   moose_test-opt                      0x000000010639c512 main + 114
8: 8   libdyld.dylib                       0x00007fff916a25ad start + 1


*** Warning ***
Processor id 0 has entered the partitioner keyword branch.
at /Users/schus/projects/moose/framework/src/mesh/MooseMesh.C, line 1801
```

**Comment by snschune** [comment]
I forgot: the `PartitionerAction` is only called once.

**Comment by permcody** [comment]
Yes, it is called twice the second is for the displaced mesh. Take a look at SetupMeshCompleteAction.C

**Comment by permcody** [comment]
Fixed by #7204
